recommendation/model/post_rec.py

```diff
@@ -78,28 +81,209 @@ def get_matrix(posts, n):
+
+    # using logs for numerical values
+    rating_features = getNumericalSimilarity(all_user_rating, 5)
+    bids_features = getNumericalSimilarity(all_bid_counts, False)
+    likes_features = getNumericalSimilarity(all_likes, False)
+
+    n_items = len(cat_features[0])
+    
+    # previous blocker where pointers were to the same object and matrix was duplicated 
+    sim_matrix = [[None for _ in range(n_items)] for _ in range(n_items)]
+
+    
+    # for each post
+    for row_idx in range(n_items):
+
+        # for each of the other posts, get calculated feature
+        for col_idx in range(n_items): 
+        
+
+            w_cat = cat_features[row_idx][col_idx]
+            w_desc = desc_features[row_idx][col_idx]
+            w_title = title_features[row_idx][col_idx]    
+            w_prompt = prompts_features[row_idx][col_idx]
+
+            w_rating = rating_features[row_idx][col_idx]
+            w_bids = bids_features[row_idx][col_idx]
+            w_likes = likes_features[row_idx][col_idx]
+
+            total_sim = (w_cat, w_desc, w_title , w_prompt, w_rating, w_bids, w_likes)
+
+            if all_posts[col_idx] is not None:                
+                sim_matrix[row_idx][col_idx] = total_sim
+            else:
+                sim_matrix[row_idx][col_idx] = []
+        
+        if row_idx == 6:                
+            weights = [0.2, 0.2, 0.2, 0.3, 0.05, 0.05, 0.05 ]
+
+            for col_idx in range(n_items):
+                total_sim = 0
+                if not sim_matrix[row_idx][col_idx]: continue                    
+
+                for i in range(len(weights)):
+                    total_sim += (sim_matrix[row_idx][col_idx][i] * weights[i])                    
+                    
+    return sim_matrix, n_items
+
+
+def getPostRecommendations(user_recs, user_likes, weights, n, post_matrix):    
+    sim_scores = []
+
+    sim_matrix = [[0 for _ in range(n)] for _ in range(n)]
+
+    # for each post
+    for row_idx in range(n):
+
+        # for each of the other posts, get calculated feature
+        for col_idx in range(n): 
+            if not post_matrix[row_idx][col_idx]: 
+                sim_matrix[row_idx][col_idx] = 0
+                continue
+
+            total_sim = 0        
+            for i in range(len(weights)):                
+                total_sim += (post_matrix[row_idx][col_idx][i] * weights[i])
+            
+            sim_matrix[row_idx][col_idx] = total_sim if total_sim <= 1 else 1
+
+    liked_ids = [like['post']['id'] for like in user_likes]
+    liked_recs = set()
+
+    threshold = 0.3
     for like in user_likes:
         # sort by similarity scores
-        sorted_sims = sorted(list(enumerate(sim_matrix[like['post']['id']])), key=lambda x: x[1], reverse=True)
+        sorted_sims = sorted(list(enumerate(sim_matrix[like['post']['id']])), key=lambda x: x[1],reverse=True)
+
+        # get only the top recommendations with threshold         
```

Comment:
great way to furhter filter down recommendations!

Should we also handle the case where user may have only 5 recs, we should give them all of them? instead of getting top 70%? 

Up to you if this is something you've considered!